scripts/get_tau_pet.py

```diff
@@ -1,94 +1,121 @@
-            dfs.append(tau)
+    # load flortaucipir spreadsheet
+    tau_df = pd.read_csv(file, index_col=0, header=0, low_memory=False)
 
-# concatenate individual tau dfs
-master_tau = pd.concat(dfs)
+    # keep only relevant columns
+    tau_df = tau_df.filter(['EXAMDATE','META_TEMPORAL_SUVR'], axis=1)
 
-# merge tau data into baseline df
-baseline = baseline.join(master_tau)
+    # convert to datetime
+    tau_df['EXAMDATE'] = pd.to_datetime(tau_df['EXAMDATE'])
 
-# create a new df with only tau values that were collected within a 12month window of the baseline visit
-baseline_tau = baseline[(baseline.EXAMDATE > baseline.date_lower) & (baseline.EXAMDATE < baseline.date_upper)]
+    # sort tau df by index and date. Also throws an error
+    tau_df.sort_values(by=['RID', 'EXAMDATE'],inplace=True)
+    
+    return (tau_df)
 
-# save df
-baseline_tau.to_csv('/Users/natashaclarke/Documents/SIMEXP/cwas_neuro_sandbox/adni_cwas/baseline_tau.csv')
\ No newline at end of file
+def match_baseline_tau(baseline_df, tau_df):
+    
+    '''
+    Group subjects in the tau df, look them up in the baseline df, and if they match return the tau
+    value closest to the session date in baseline. Return a list of dfs, one per subject, merge into
+    baseline df, and then keep only those within a 12 month window. This seems a silly and expensive way 
+    to do this! But, I ran into all sorts of problems with the datetime format and this is the only way 
+    I figured it out...
+    '''
+    
+    tau_dfs_list = []
+    for tau_id, group in tau_df.groupby(level='RID'):
+        for baseline_id, session in zip(baseline_df.index, baseline_df.session):
+            if tau_id == baseline_id:
+                participant_df = group
+                participant_df.set_index('EXAMDATE', inplace=True)
+                participant_tau = pd.DataFrame(participant_df['META_TEMPORAL_SUVR'][participant_df.index[[participant_df.index.get_loc(session, method='nearest')]]])
+                participant_tau['EXAMDATE'] = participant_tau.index
+                participant_tau.index = [baseline_id]
+                participant_tau.rename_axis('RID',inplace=True)
+                tau_dfs_list.append(participant_tau)
+                
+    # concatenate individual tau dfs
+    master_tau = pd.concat(tau_dfs_list)
+    
+    # merge tau data into baseline df
+    baseline_df = baseline_df.join(master_tau)
+    
+    # create a new df with only tau values that were collected within a 12month window of the baseline visit
+    baseline_tau = baseline_df[(baseline_df.EXAMDATE > baseline_df.date_lower) & (baseline_df.EXAMDATE < baseline_df.date_upper)]
+                
+    return (baseline_tau)
+
+# run functions
+baseline_df = get_baselines(adni_data)
+tau_df = get_tau(tau_data)
+baseline_tau = match_baseline_tau(baseline_df, tau_df)
+
+# save baseline and matched tau data in data directory
+baseline_tau.to_csv(Path("__file__").parent.parent / 'data' / 'baseline_tau.csv')
```

Comment:
```suggestion
if __name__ == '__main__':
    # run functions
    baseline_df = get_baselines(adni_data)
    tau_df = get_tau(tau_data)
    baseline_tau = match_baseline_tau(baseline_df, tau_df)
    
    # save baseline and matched tau data in data directory
    baseline_tau.to_csv(Path("__file__").parent.parent / 'data' / 'baseline_tau.csv')

```

This will allow you to reuse the three functions through `import` in other script without running this script.